arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep

```diff
@@ -30,148 +45,56 @@ param workspaceResourceId string
-
-@description('Optional. Number of threadshold violation to trigger the alert')
-@minValue(0)
-@maxValue(10000)
-param breachesThreshold int = 3
-
-@description('Optional. The list of actions to take when alert triggers.')
+@description('Optional. Actions to invoke when the alert fires.')
 param actions array = []
 
-@description('Optional. The list of action alert creterias.')
-param criterias array = []
-
-@description('Optional. Type of the alert criteria.')
-@allowed([
-  'AlertingAction'
-  'LogToMetricAction'
-])
-param odataType string = 'AlertingAction'
+@description('Optional. The rule criteria that defines the conditions of the scheduled query rule.')
+param criterias object = {}
 
-@description('Optional. Suppress Alert for (in minutes).')
-param suppressForMinutes int = 0
+@description('Optional. Mute actions for the chosen period of time (in ISO 8601 duration format) after the alert is fired. If set, autoMitigate must be disabled.Relevant only for rules of the kind LogAlert.')
+param suppressForMinutes string = ''
 
 @description('Optional. Tags of the resource.')
 param tags object = {}
 
 @description('Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered')
 param cuaId string = ''
 
-var metricTrigger = {
-  metricColumn: metricColumn
-  metricTriggerType: breachesTriggerType
-  threshold: breachesThreshold
-  thresholdOperator: breachesThresholdOperator
-}
-
 module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
   name: 'pid-${cuaId}'
   params: {}
 }
 
-resource queryAlert 'microsoft.insights/scheduledQueryRules@2018-04-16' = {
+resource queryAlert 'Microsoft.Insights/scheduledQueryRules@2021-02-01-preview' = {
   name: alertName
   location: location
   tags: tags
+  kind: kind
   properties: {
+    actions: {
+      actionGroups: actions
+      customProperties: {}
+    }
+    autoMitigate: contains(kind, 'LogAlert') ? autoMitigate : null
+    criteria: criterias
+
     description: alertDescription
+    displayName: alertName
     enabled: enabled
-    source: {
-      query: query
-      authorizedResources: authorizedResources
-      dataSourceId: workspaceResourceId
-      queryType: 'ResultCount'
-    }
-    schedule: {
-      frequencyInMinutes: evaluationFrequency
-      timeWindowInMinutes: windowSize
-    }
-    action: {
-      severity: severity
-      aznsAction: {
-        actionGroup: actions
-      }
-      throttlingInMin: suppressForMinutes
-      trigger: {
-        thresholdOperator: metricResultCountThresholdOperator
-        threshold: metricResultCountThreshold
-        metricTrigger: (empty(metricColumn) ? null : metricTrigger)
-      }
-      criteria: criterias
-      'odata.type': 'Microsoft.WindowsAzure.Management.Monitoring.Alerts.Models.Microsoft.AppInsights.Nexus.DataContracts.Resources.ScheduledQueryRules.${odataType}'
-    }
+    evaluationFrequency: contains(kind, 'LogAlert') ? evaluationFrequency : null
```

Comment:
I'd like to understand why this is implemented like this. 

1. Why using this logic and not just reference `evaluationFrequency`?
2. Why contains?

Reply:
The docs state, that only when the alert is of type LogAlert this parameter should be used.

Reply:
alright. However, I guess we should change it to 
```bicep
evaluationFrequency: (kind == 'LogAlert') ? evaluationFrequency : null
```

Reply:
I like it and I changed it ;)